[UWP] FormsTextBox clean up OnApplyTemplate

bmacombe opened the pull request:
### Description of Change ###

Changing to safe casting when pulling elements out of the control template incase the template changes.  Per a suggestion in this comment on PR https://github.com/xamarin/Xamarin.Forms/pull/11140#discussion_r448810684

I also removed the code pulling out the BorderElement from the template.  The variable was not referenced, other than to be set and windows phone is no longer supported.


### Issues Resolved ### 
<!-- Please use the format "fixes #xxxx" for each issue this PR addresses -->

None, housekeeping only

### API Changes ###
 
 None

### Platforms Affected ### 
<!-- Please list all platforms affected by these changes -->
- UWP

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###
I retest all issues from PR 11140 and all tested good.

### PR Checklist ###
<!-- To be completed by reviewers -->

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

samhouts commented:
@bmacombe I just cherry-picked #11140 to 4.7.0, so would you please rebase this PR on that branch? Thanks!!

bmacombe commented:
@samhouts Done!  Glad to see #11140 going back to 4.7 Thanks!

I also moved a comment explaining what 11140 was doing to the correct place, I noticed I moved the code, but not the comment

samhouts commented:
@bmacombe Thanks so much! We appreciate you!!

bmacombe commented:
@samhouts You will need to cherry pick de519abb75ec47db2d21d7f217d99b7c37d45746 into 4.7 also

samhouts commented:
@bmacombe Good call! I thought that was part of the original PR. Done!

bmacombe commented:
@samhouts Should have been part of it! :)

samhouts commented on Xamarin.Forms.Platform.UAP/FormsTextBox.cs:
```diff
@@ -157,14 +156,7 @@ protected override void OnApplyTemplate()
 		{
 			base.OnApplyTemplate();
 
-			if (Device.Idiom == TargetIdiom.Phone)
```
😭 still too soon

hartez replied:
😢

bmacombe replied:
I with you! loved Windows Phone!

bmacombe commented:
@samhouts  I rebased on 4.7 to fix the build fail

bmacombe commented:
This PR should be merged after #11351 now.  I rebased on it that branch because of some comment updates on it.

bmacombe commented:
Rebased on 4.7 now that #11351 is merged
